lib/code_fund/properties/properties.ex

```diff
@@ -284,12 +283,17 @@ defmodule CodeFund.Properties do
     do: Repo.get_by!(Property, name: name) |> Repo.preload([:user, :audience])
 
   def get_all_display_rates(%Property{} = property) do
-    AdService.Query.ForDisplay.build(
-      programming_languages: property.programming_languages,
-      topic_categories: property.topic_categories
-    )
-    |> CodeFund.Repo.all()
-    |> AdService.Math.Basic.get_all_display_rates()
+    audience = property.audience
+
+    case audience do
+      nil ->
+        []
+
+      %Audience{} = audience ->
+        AdService.Query.ForDisplay.build(audience, nil)
```

Comment:
If the old `build` function based on programming languages and topic categories isn't called any more, can we rip out all of that unused logic/function calls?

Reply:
Yes